parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -342,51 +349,69 @@ pub mod pallet {
 		}
 
 		#[pallet::call_index(3)]
-		#[pallet::weight(1000)]
+		#[pallet::weight(1_000_000_000)]
+		#[transactional]
+		pub fn block_bridge(origin: OriginFor<T>) -> DispatchResult {
+			let _sender = ensure_root(origin)?;
+
+			<Blocked<T>>::set(true);
+
+			log::info!(target: "ethereum-beacon-client","💫 bridge emergency blocked from governance.");
+
+			Self::deposit_event(Event::BridgeBlocked { blocked: true });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(10_000_000_000)]
 		#[transactional]
-		pub fn unblock_bridge(origin: OriginFor<T>) -> DispatchResult {
+		pub fn unblock_bridge(
+			origin: OriginFor<T>,
+			check_points: Option<Vec<CheckpointSyncOf<T>>>,
+		) -> DispatchResult {
 			let _sender = ensure_root(origin)?;
 
+			log::info!(target: "ethereum-beacon-client","💫 unblock bridge from governance with provided checkpoints.");
+			if check_points.is_some() {
+				for check_point in check_points.unwrap() {
+					Self::process_checkpoint_sync(check_point)?;
```

Comment:
`process_checkpoint_sync` uses `verify_sync_committee` to verify checkpoint. My understanding is that it validates that the provided SyncCommittee exists within the beacon state at the header specified.

We provide a list of CheckPoints in `unblock_bridge` to process in order. It looks like we validate each checkpoint independently. We do not explicitly check that all these checkpoints lie on the canonical chain or any SyncCommitee handovers. Is this correct?

Will this be handled by governance in the sense that the community would make sure the checkpoints are correct before voting on a proposal that would enact the unblocking?

Reply:
Exactly!